test/jsonapi/serializer_test.exs

```diff
@@ -536,39 +535,31 @@ defmodule JSONAPI.SerializerTest do
     Application.delete_env(:jsonapi, :remove_links)
   end
 
-  test "serialize includes pagination links if they are defined and with_pagination is configured" do
-    data = %{id: 1}
-    Application.put_env(:jsonapi, :with_pagination, true)
+  test "serialize includes pagination links if page-based pagination is requested" do
+    data = [%{id: 1}]
 
-    encoded = Serializer.serialize(PostView, data, nil)
+    conn =
+      %Plug.Conn{}
+      |> Plug.Conn.assign(:jsonapi_query, %JSONAPI.Config{
```

Comment:
Let's not use a pipeline here since it's only a single function.

Reply:
I refactored the test to use Plug.Test.conn/3